src/core/evaluator.js

```diff
@@ -167,6 +167,35 @@ function normalizeBlendMode(value, parsingArray = false) {
   return "source-over";
 }
 
+function getFallbackEncoding(properties) {
+  // According to "Table 114" in section "9.6.6.1 General" (under
+  // "9.6.6 Character Encoding") of the PDF specification, a Nonsymbolic
+  // font should use the `StandardEncoding` if no encoding is specified.
+  const isSymbolicFont = !!(properties.flags & FontFlags.Symbolic);
+  const isNonsymbolicFont = !!(properties.flags & FontFlags.Nonsymbolic);
+
+  // The Symbolic attribute can be misused for regular fonts
+  // Heuristic: we have to check if the font is a standard one also
+  if (isSymbolicFont) {
+    if (!properties.file || properties.isInternalFont) {
+      if (/Symbol/i.test(properties.name)) {
+        return SymbolSetEncoding;
+      } else if (/Dingbats|Wingdings/i.test(properties.name)) {
+        return ZapfDingbatsEncoding;
+      }
+    }
+    if (properties.type === "TrueType" && !isNonsymbolicFont) {
+      // Use the font-name to try and guess if `WinAnsiEncoding` is more
+      // appropriate than `MacRomanEncoding` (fixes issue15910.pdf).
+      if (/Arial/i.test(properties.name)) {
```

Comment:
I really don't know either; since any name-matching (no matter from where) seem somewhat arbitrary and thus risky.

The reason that I picked the one from the PDF document, rather than the font, was mostly that it simplified the overall patch by not requiring a bunch of new code in the `src/core/fonts.js` file just for this one broken PDF document :-)

Reply:
And still about the `name` table, there is:
```xml
<name>
    <namerecord nameID="1" platformID="3" platEncID="1" langID="0x409">
      Arial
    </namerecord>
    <namerecord nameID="2" platformID="3" platEncID="1" langID="0x409">
      Regular
    </namerecord>
    <namerecord nameID="3" platformID="3" platEncID="1" langID="0x409">
      Arial
    </namerecord>
    <namerecord nameID="4" platformID="3" platEncID="1" langID="0x409">
      Arial
    </namerecord>
    <namerecord nameID="5" platformID="3" platEncID="1" langID="0x409">
      Version 1.00
    </namerecord>
    <namerecord nameID="6" platformID="3" platEncID="1" langID="0x409">
      Arial
    </namerecord>
  </name>
```

According to:
https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6name.html#ID
`platformID === 3` is for Microsoft encoding.

I changed the strings like `/UIYZDX+Arial` to `/UIYZDX+xrial` in the pdf and it's still ok in Acrobat.
I don't know how to replace the font by a modified one in the pdf so I can't verify that changing the name in the `name` has an effect or not.
Anyway I'd say that it's less risky to rely on that, if it's correct of course, than on the name.